src/app/cases/case_components/case-therapy/case-therapy.component.ts

```diff
@@ -1,8 +1,102 @@
 import { Component } from '@angular/core';
+import { FormBase } from '../../../shared/dynamic-form/types/form-element-base';
+import { SentResourceTypes } from '../../../_constants/common';
+import { CaseDataDto, PrescriptionDto, TherapyDto, TreatmentDto } from '../../../_models/models';
+import { BaseService } from '../../../_services/api/base.service';
+import { PrescriptionService } from '../../../_services/api/prescription.service';
+import { TreatmentService } from '../../../_services/api/treatment.service';
+import { FormElementControlService } from '../../../_services/form-element-control.service';
+import { NotificationService } from '../../../_services/notification.service';
+import { SendResourceService } from '../../../_services/send-resource.service';
+import * as data from './case-therapy-form-data';
 
 @Component({
   selector: 'app-case-therapy',
   templateUrl: './case-therapy.component.html',
   styleUrls: ['./case-therapy.component.scss'],
 })
-export class CaseTherapyComponent {}
+export class CaseTherapyComponent {
+  therapy: TherapyDto;
+  prescription: PrescriptionDto;
+  treatment: TreatmentDto;
+  myFormElements: FormBase<any>[] = [];
+  formData = data.FORM_DATA_CASE_THERAPY;
+
+  public resourceService: BaseService<any>;
+
+  constructor(
+    private formElementControlService: FormElementControlService,
+    private prescriptionService: PrescriptionService,
+    private treatmentService: TreatmentService,
+    private notificationService: NotificationService,
+    private sendResourceService: SendResourceService
+  ) {}
+
+  setPrescriptionFormData(): void {
+    this.myFormElements = this.formElementControlService.setValuesForDynamicForm(
+      this.prescription,
+      this.formData
+    );
+  }
+
+  setTreatmentFormData(): void {
+    this.myFormElements = this.formElementControlService.setValuesForDynamicForm(
+      this.treatment,
+      this.formData
+    );
+  }
```

Comment:
This can not be right.
You are assigning this this,myFormElements
    this.myFormElements = this.formElementControlService.setValuesForDynamicForm(
      this.prescription,
      this.formData
    );

and in the function bellow you are overwriting the same variable 
this.myFormElements = this.formElementControlService.setValuesForDynamicForm(
      this.treatment,
      this.formData
    );

Reply:
Resolved with 2 forms